Update translations from transifex

gravitystorm opened the pull request:
This builds on #546 

This PR contains updated translations from Transifex. It also includes changes to the source translation files (i.e. the `*en-GB.yml` files) so that everything is nice and happy with the round-trips. Most of the changes in this pull request are stripping unnecessary quotes from the yml files.

We've also had the site partially translated into Italian (it) on transifex, so this is added to the list of supported locales.

nikolai-b commented on config/locales/forms.it.yml:
```diff
@@ -0,0 +1,169 @@
+          title: f5f791770fc3a0079f829fd27ff3dba5_tr
+        create: *id001
+        edit: *id001
+        update: *id001
```
Are these correct?

gravitystorm replied:
Good spot - these are clearly wrong. :-( I'll investigate what's happened here - either a bug in pulling the translations, or perhaps garbage translations within transifex.

nikolai-b commented:
I've not yet been introduced to transifex so sorry if I'm asking basic things here.  Does it make the translations easier to deal with?  Is there a simple process to pull the changes someone adds there? This seems to pull in some weird strings...

gravitystorm commented:
> I've not yet been introduced to transifex so sorry if I'm asking basic things here.

No problem, I realise it's not documented and hasn't been used in a while

> Does it make the translations easier to deal with? Is there a simple process to pull the changes someone adds there?

Yes, it makes it much easier for translators and practically no difference for developers. The translations are done through a web interface, so can be carried out without hand-editing yml files. The interface also supports things like progress meters, notifications to translators of new strings, automatic suggestions for translations and similar things. It widens the pool of translators enormously.

The process for developers is straightforward. When new strings are added to en-GB.yml a "push" command is run to update the master translation on transifex. At any time, a "pull" command can be run to download new translations from transifex into the repo, and these can then be committed. 

There's a couple of bits of grit in the system at the moment (e.g. https://github.com/cyclestreets/cyclescape/issues/548 ) that I hope to work on soon. I'm also aware that it will work better if it is used more often, so I'm happy to push/pull translations more frequently (and deal with any translation-related issues) until the process becomes more regular.

gravitystorm commented:
I've fixed the bogus strings in the source file. The problem is that transifex isn't entirely happy with the references and anchors that we're using, so I've emailed their support for help. We'll need the bug fixed at transifex before I can pull the fixed translations into this PR.

gravitystorm commented:
As an update, I've had a few emails back and forward with transifex while trying to sort this out. In summary, they acknowledged the problem, thought they'd fixed it, but turns out it's still broken. The latest (as of yesterday) is:

> I checked this out and it seems that the issue is related only to Alias indicators (*). I didn't case across any issues in case of Anchor properties.
>
> We have already added this to our backlog in order to further investigate this issue.
>
> Unfortunately, I cannot provide you with an ETA at the moment but I will definitely get back to you when any new information arises. 

My next strategy is going to be to develop a workflow which takes the source yaml, flattens out the anchors and references, and uploads this to transifex. This won't impact the translators much, since the identical strings can be re-translated by pushing a button. The downloaded translations won't contain anchors but we're not hand-editing them. I hope to complete this soon, and we can get cracking with the translations.

nikolai-b commented:
Thanks for the update!

gravitystorm commented:
I've implemented rake tasks for pushing and pulling from transifex. These tasks paper over the cracks involved in:

* transifex not liking anchors and labels - these are preserved in the en-GB originals
* transifex using underscores instead of hyphens in the language names.

So everything works smoothly. I've also added a readme to explain what's going on.

This PR is ready for review.

gravitystorm commented:
Any chance of a review on this please?

nikolai-b commented:
@gravitystorm thanks and apologies for missing this.  I've read through the documentation, does @PetrDlouhy know to only make changes on transifex?  
This means that I don't have to change any of the other languages, just push and pull regularly, yay!  The downside is that any changes that have been made locally https://github.com/cyclestreets/cyclescape/pull/547/files#diff-1024427eb018577ee278444ff0b587abR12 won't be included but they aren't too bad and that seems a price worth paying.  Code wise it makes lots of sense but as I do very little of the translation I'd just like to confirm with @PetrDlouhy is happy with this workflow.

nikolai-b commented:
@PetrDlouhy any thoughts?

mvl22 commented:
@PetrDlouhy Are you happy with this move to using Transifex now for managing translation?

Nikolai and I are keen to get this merged in ASAP.

PetrDlouhy commented:
OK, I can primarily put my translations to Transifex, no problem. Especially if you can quickly propagate the changes to the repository.

The only issue I see is, that sometimes the string was missing in English locale (was generated from the name of variable). I can't add that to Transifex, so I would need to add that to English locale first, which is a bit complicated.

I have already translated most of the strings there, so you can pull that, if you haven't done it already.

gravitystorm commented:
> that sometimes the string was missing in English locale (was generated from the name of variable).

That should no longer happen - we now have [`config.action_view.raise_on_missing_translations = true`](https://github.com/cyclestreets/cyclescape/blob/4cd99afaf2f84dc8603360aef7ef24e9e13ec62e/config/environments/development.rb#L43) set in development and testing so every translation should have an entry in the English locale.

nikolai-b commented:
@gravitystorm 

```
$ bundle exec rake transifex:push
Exception: Only maintainers are allowed to update the source file.
```

Would you mind adding me?  I've joined as a Cz translator.  My username is nikolaib

gravitystorm commented:
> Would you mind adding me? I've joined as a Cz translator. My username is nikolaib

I've added you as a maintainer for the project, and also as an administrator for the overall organisation.
